### DISCUSSION TITLE
Custom roles & access for external users

### DISCUSSION THREAD

**Discussion by ono-mk** [Budibase/budibase#3922] [Help]
Hi All - I'm evaluating using Budibase to build a vendor panel for a small cohort of external users, and trying to understand the permissioning. Ideally, the vendors will only be able to view records that are theirs (ex: for e-commerce, vendor should only be able to view their own SKUs & orders). Is there any documentation you can point me to around how to implement this? We don't want vendors to be able to view the details for other vendors.

**Comment by mjashanks** [comment]
So, assuming that your vendors are users, I would add a "Vendor" column to your user table. This would be a `Text` or `Options` type.

Then, each SKU would also have a "Vendor" field.

When listing items for a particular user, you would then put a filter on your data provider like:

`Vendor Equals binding {{ Current User.Vendor }}`

I assume that you would also want to assign this screen to a "Vendor" role. You can create new roles under "Manage Roles", in the Users table.
